shell/platform/android/io/flutter/plugin/text/ProcessTextPlugin.java

```diff
@@ -0,0 +1,175 @@
+
+  @NonNull
+  private Map<Integer, MethodChannel.Result> requestsByCode =
+      new HashMap<Integer, MethodChannel.Result>();
+
+  public ProcessTextPlugin(@NonNull ProcessTextChannel processTextChannel) {
+    this.processTextChannel = processTextChannel;
+    this.packageManager = processTextChannel.packageManager;
+
+    this.cacheResolveInfos();
+
+    processTextChannel.setMethodHandler(
+        new ProcessTextChannel.ProcessTextMethodHandler() {
+          @Override
+          public Map<Integer, String> queryTextActions() {
+            Map<Integer, String> result = new HashMap<Integer, String>();
+            for (Integer id : resolveInfosById.keySet()) {
+              final ResolveInfo info = resolveInfosById.get(id);
+              result.put(id, info.loadLabel(packageManager).toString());
+            }
+            return result;
+          }
+
+          @Override
+          public void processTextAction(
+              @NonNull int id,
+              @NonNull String text,
+              @NonNull boolean readOnly,
+              @NonNull MethodChannel.Result result) {
+            if (activityBinding == null) {
+              result.error("error", "Plugin not bound to an Activity", null);
+              return;
+            }
+
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+              result.error("error", "Android version not supported", null);
+              return;
+            }
+
+            final ResolveInfo info = resolveInfosById.get(id);
+            if (info == null) {
+              result.error("error", "Text processing activity not found", null);
+              return;
+            }
+
+            Integer requestCode = result.hashCode();
+            requestsByCode.put(requestCode, result);
+
+            Intent intent =
+                new Intent()
+                    .setClassName(info.activityInfo.packageName, info.activityInfo.name)
+                    .setAction(Intent.ACTION_PROCESS_TEXT)
+                    .setType("text/plain")
+                    .putExtra(Intent.EXTRA_PROCESS_TEXT, text)
+                    .putExtra(Intent.EXTRA_PROCESS_TEXT_READONLY, readOnly);
+
+            // Start the text processing activity. onActivityResult callback is called
+            // when the activity completes.
+            activityBinding.getActivity().startActivityForResult(intent, requestCode);
+          }
+        });
+  }
+
+  private void cacheResolveInfos() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+      return;
+    }
+
+    Intent intent = new Intent().setAction(Intent.ACTION_PROCESS_TEXT).setType("text/plain");
+
+    List<ResolveInfo> infos;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      infos = packageManager.queryIntentActivities(intent, PackageManager.ResolveInfoFlags.of(0));
+    } else {
+      infos = packageManager.queryIntentActivities(intent, 0);
```

Comment:
The previous version is deprecated, the new one allow using 'long' values instead of 'int' for the previous one.
I will probabl add a comment above the call to the old one to mention it is deprecated.